### ISSUE TITLE
a string that refers to a store path cannot be appended to a path

### ISSUE THREAD

**Issue by LEXUGE** [cargo2nix/cargo2nix#233]
```
error: a string that refers to a store path cannot be appended to a path

       at /nix/store/qsbymj5zlc7z5fnb9sg0b3kva803mjrl-source/overlay/lib/fetch.nix:16:7:

           15|     in
           16|       /. + builtins.readFile (runCommand "find-crate-${name}-${version}"
             |       ^
           17|         { nativeBuildInputs = [ jq remarshal ]; }
(use '--show-trace' to show detailed location information)
```
The exact copy of code works on my local machine with `nix-2.6.0pre20211217_6e6e998`
The above error is origninated from `nix 2.6.0`. I think we would be better off to use `builtins.toPath` instead of `/.`

**Comment by LEXUGE** [comment]
https://github.com/flibrary/cargo2nix/commit/e2a4c33f1014042bddadfc36a8be6b3ba7964955
This seems to resolve the issue by doing a workaround mentioned https://discourse.nixos.org/t/what-am-i-doing-wrong-here/2517/3

**Comment by moreSocratic** [comment]
Is this fix able to be made into a PR against cargo2nix, or is it a temporary workaround?

**Comment by LEXUGE** [comment]
It is a workaround as I consider it to be hacky. However, it works just fine.

**Comment by psionic-k** [comment]
Well now I'm trying to figure out what this hunk needs to do and how to do it more correctly.  Also ran into this issue.  Git repos broken in #224

**Comment by psionic-k** [comment]
Basically it looks like we want to get rid of this `readFile`.  What's going on here is that we download a git source and don't know where the crate is.

We can do this at build time.  There's no reason to check for the crate before trying to build since we can just find it during the build phase.  Pushes errors later in the build, but removes import-from-derivation nastiness.

**Comment by psionic-k** [comment]
I got to work on this.  Would you believe that `remarshal` is still broken when reading many tomls?  This means the `runCommand` would break if the git repo used one of the offending `Cargo.toml` patterns.

I have it working, just considering doing something better than using the `sed` and `grep` hack on every single toml in every workspace just to look for the crate we want to build.

**Comment by psionic-k** [comment]
I think that https://github.com/cargo2nix/cargo2nix/pull/224/commits/92f0b73b197380468a1a2430c5afeef8d189d73f  will fix this issue.  Little bit hesitant about doing this for every crate.   Hoping to get rid of remarshal by using cargo metadata and jq.

**Comment by psionic-k** [comment]
Proposed fix is in #224  and will close this issue on merge.

Okay, dropped metadata pretty fast since, as others had said earlier, cargo doesn't know what `--offline` should means.

I resorted to a simple grep and some bash logic.  This is a heuristic solution and may fail in some odd cases where someone manages to get a `name = "something"` key written in an unexpected `Cargo.toml`
